js/data/symbol_bucket.js

```diff
@@ -40,6 +40,8 @@ function SymbolBucket(options) {
 
     this.adjustedIconMaxSize = this.layer.getLayoutValue('icon-size', 18, zoomHistory);
     this.adjustedIconSize = this.layer.getLayoutValue('icon-size', this.zoom + 1, zoomHistory);
+
+    this.index = 0;
```

Comment:
Could we get away with passing `k` to [`addFeature` here](https://github.com/mapbox/mapbox-gl-js/pull/2023/files#diff-ce6c52178591c2d5fe623678de7944e5R228) instead? I'd prefer that to adding another class property.